app/utils/pages.tsx

```diff
@@ -13,61 +10,78 @@ interface IPage {
+    const hasContestantInfo = !!showContestantData && showContestantData.length;
+    if(hasContestantInfo){
+        const scoringSubpage = {
+            name: "Scoring",
+            path: `/${pageInformation.showStatus}/${pageInformation.showNameAndSeason}/scoring`
+        }
+        const leagueStandingSubpage = {
+            name: "League Standing",
+            path: `/${pageInformation.showStatus}/${pageInformation.showNameAndSeason}/league-standing`
+        }
+        subpages.push(scoringSubpage, leagueStandingSubpage);
+    }
+    return subpages;
 }
 
-export function checkForSubpages (filename:string){
-    return fs.existsSync(path.join(process.cwd(), "app", "leagueData", filename));
+function generatePathObj(pageData:PageInformation, showContestantData:IContestantData[]){
+    const subpages:Array<IPage> = generateSubpages(pageData, showContestantData);
+    if(pageData.showStatus === "active"){
+        pageData.friendlyName = `Current (${pageData.friendlyName})`
+    }
+    return {
+        name: pageData.friendlyName,
+        subpages: subpages
+    }
 }
 
-export function getPages(): ILeagueLink[] {
+export async function getPages(): Promise<ILeagueLink[]> {
     // Based on availability in leagueConfiguration
-    const filepaths = pagesModule.getPathsToMap();
+    const leagueConfigurationKeys = await getLeagueConfigurationKeys();
     const activeLeaguePaths:Array<ILeagueLink> = [];
     const archiveLeaguePaths:Array<ILeagueLink> = [];
-    filepaths.map((file: string) => {
-        // Needed status for url
-        const { leagueStatus } = pagesModule.getLeagueConfigurationData(file);
-        // Parses filename and converts it to url format
-        const pageStrings = transformFilenameToSeasonNameRepo(file);
-        const subpages:Array<IPage> = [];
-        subpages.push({
-            name: "Contestants",
-            path: `/${leagueStatus}/${pageStrings.urlSlug}/contestants`
-        });
-        if(pagesModule.checkForSubpages(file)){
-            const scoringSubpage = {
-                name: "Scoring",
-                path: `/${leagueStatus}/${pageStrings.urlSlug}/scoring`
-            }
-            const leagueStandingSubpage = {
-                name: "League Standing",
-                path: `/${leagueStatus}/${pageStrings.urlSlug}/league-standing`
-            }
-            subpages.push(scoringSubpage, leagueStandingSubpage);
-        }
-        if(leagueStatus === "active"){
-            pageStrings.friendlyName = `Current (${pageStrings.friendlyName})`
-        }
-        //   Path object created
-        const pathObj = {
-            name: pageStrings.friendlyName,
-            subpages: subpages
-        }
-        if(leagueStatus === "active"){
+    for(const leagueConfigurationKey of leagueConfigurationKeys){
+        const pageData:PageInformation = constructPageInformation(leagueConfigurationKey);
+        const showContestantData = await getContestantData(pageData.contestantDataKey);
```

Comment:
As we talked about, I think we should probably not do this call (which will be at least a half a dozen queries), instead we should write another lil function which just gets a single cursors worth to check for the existence of contestant data. If you want to then use that new function inside of `getContetantData` that would be a good idea, and if you would like to do all this in another PR, I would do that first and come back to this to work that in.

Reply:
Committed a fix addressing this :)